src/libraries/System.Private.CoreLib/generators/EventSourceGenerator.Emitter.cs

```diff
@@ -12,76 +12,72 @@ namespace Generators
 {
     public partial class EventSourceGenerator
     {
+        /// <summary>Code for a [GeneratedCode] attribute to put on the top-level generated members.</summary>
+        private static readonly string s_generatedCodeAttribute = $"[global::System.CodeDom.Compiler.GeneratedCodeAttribute(\"{typeof(EventSourceGenerator).Assembly.GetName().Name}\", \"{typeof(EventSourceGenerator).Assembly.GetName().Version}\")]";
+        /// <summary>Header comments and usings to include at the top of every generated file.</summary>
+        private static readonly string s_header =
+            "// <auto-generated/>" + Environment.NewLine;
+
         private sealed class Emitter
         {
-            private readonly StringBuilder _builder = new StringBuilder(1024);
-            private readonly GeneratorExecutionContext _context;
-
-            public Emitter(GeneratorExecutionContext context) => _context = context;
-
-            public void Emit(EventSourceClass[] eventSources, CancellationToken cancellationToken)
+            public static void Emit(SourceProductionContext spc, EventSourceClass ec)
             {
-                foreach (EventSourceClass? ec in eventSources)
-                {
-                    if (cancellationToken.IsCancellationRequested)
-                    {
-                        // stop any additional work
-                        break;
-                    }
+                StringBuilder sb = new StringBuilder(1024);
 
-                    _builder.AppendLine("using System;");
-                    GenType(ec);
+                sb.AppendLine(s_header);
```

Comment:
Why put s_header in a static rather than just having this be: `sb.AppendLine("// <auto-generated/>").AppendLine();`?

Reply:
I followed what other source generators with more complex headers were doing, but I made it simpler.